src/System.Drawing.Common/src/System/Drawing/Printing/InvalidPrinterException.Serializable.cs

```diff
@@ -3,27 +3,20 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Runtime.Serialization;
-using System.Security.Permissions;
 
 namespace System.Drawing.Printing
 {
-    [Serializable]
     partial class InvalidPrinterException
     {
         protected InvalidPrinterException(SerializationInfo info, StreamingContext context) : base(info, context)
         {
             _settings = (PrinterSettings)info.GetValue("settings", typeof(PrinterSettings));
         }
 
-        [SecurityPermissionAttribute(SecurityAction.Demand, SerializationFormatter = true)]
         public override void GetObjectData(SerializationInfo info, StreamingContext context)
         {
-            if (info == null)
-            {
-                throw new ArgumentNullException("info");
-            }
-            info.AddValue("settings", _settings);
             base.GetObjectData(info, context);
+            info.AddValue("settings", _settings);            
```

Comment:
I don't see this file in the csproj - am I missing it? I don't know why they are split into 3 in the first place

Reply:
Mono is using the whole System.Drawing.Common package. As they need to support more serialization types those files got split up. The InvalidPrinterException.Serializable.cs is included by Mono.

I updated the files as I forgot a TypeForward (I can't test that type) and renamed them.